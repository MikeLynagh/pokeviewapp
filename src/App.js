import React, { Component, useState } from "react";
import Header from "./component/Header";
import { Route, Routes, BrowserRouter, Switch } from "react-router-dom";
import SearchApp from "./component/SearchApp";
import Books from "./component/Pokedex/PokemonList";
import About from "./component/About";
import Login from "./component/Login/Login";
import useToken from "./component/Login/useToken";


function App(){
    const { token, setToken } = useToken()

    // a token must be return otherwise, the web application will display the login page. 



    if(!token){
        return <Login setToken={setToken} />
    }
    return(
        <React.Fragment>
            <header>
            <Header />
            </header>
            <main>
                <Routes>
                    <Route path="/" element={<SearchApp />} exact />
                    <Route path="/about" element={<About />} exact />

                </Routes>

            </main>
        </React.Fragment>
    )
}



export default App