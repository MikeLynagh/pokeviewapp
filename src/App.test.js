import { render, screen, cleanup } from '@testing-library/react';
import React from 'react';
import App from './App';

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

afterEach(cleanup)
 
 it('should take a snapshot', () => {
    const { asFragment } = render(<App />)
    
    expect(asFragment(<App />)).toMatchSnapshot()
   })