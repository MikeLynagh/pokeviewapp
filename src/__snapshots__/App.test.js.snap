// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="logo"
    >
      <img
        alt="Pokemon Go logo png"
        src="https://www.freepnglogos.com/uploads/pok-mon-go-logo-png-30.png"
      />
    </div>
    <div
      class="text-center mt-4 name"
    >
      Pokedex
    </div>
    <h1>
      Please Log In
    </h1>
    <form
      class="p-3 mt-3"
    >
      <div
        class="form-field d-flex align-items-center"
      >
        <span
          class="far fa-user"
        />
        <input
          id="userName"
          name="userName"
          placeholder="Username"
          type="text"
        />
      </div>
      <div
        class="form-field d-flex align-items-center"
      >
        <span
          class="fas fa-key"
        />
        <input
          id="pwd"
          name="password"
          placeholder="Password"
          type="password"
        />
      </div>
      <div>
        <button
          class="btn mt-3"
          type="submit"
        >
          Login
        </button>
      </div>
      <div>
        <p>
          You must use the username: admin
        </p>
        <p>
          You must use the password: admin
        </p>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
